d3a_interface/sim_results/simulation_assets_info.py

```diff
@@ -0,0 +1,43 @@
+from typing import Dict, List
+
+from d3a_interface.sim_results.results_abc import ResultsBaseClass
+
+
+class SimulationAssetsInfo(ResultsBaseClass):
+    FIELDS = ['number_of_house_type', 'avg_assets_per_house', 'number_of_load_type',
+              'total_energy_demand_kwh', 'number_of_pv_type', 'total_energy_generated_kwh',
+              'number_of_storage_type', 'total_energy_capacity_kwh', 'number_of_power_plant_type',
+              'max_power_plant_power_kw']
+
+    def __init__(self):
+        for field in SimulationAssetsInfo.FIELDS:
+            setattr(self, field, 0)
+
+    @staticmethod
+    def merge_results_to_global(market_results: Dict, global_results: Dict, slot_list: List):
+        pass
+
+    def update(self, area_result_dict, core_stats, current_market_slot):
+        updated_results_dict = {field: 0 for field in SimulationAssetsInfo.FIELDS}
+        for area_uuid, area_result in core_stats.items():
+            if 'total_energy_demanded_wh' in area_result:
+                updated_results_dict['number_of_load_type'] += 1
+                updated_results_dict['total_energy_demand_kwh'] += area_result['total_energy_demanded_wh']/ 1000.0
+            elif 'pv_production_kWh' in area_result:
+                updated_results_dict['number_of_pv_type'] += 1
+                updated_results_dict['total_energy_generated_kwh'] += area_result['pv_production_kWh']
+            elif 'Storage' in area_result:  # Needs to update the result sent from d3a to send the capacity
```

Comment:
AFAIU, only the capacity of the storage is needed, which will only change with live events. The capacity is already in the configuration No need to send it again from the d3a.

Reply:
IIRC latest configuration is updated immediately once the live event is created and sent to d3a, and not when it is really applied to d3a. This means that the latest configuration contains values that have not affected the simulation yet. For normal simulations this is not a problem, because the slots are executed fast enough for the user to not notice. However, for CNs, this process might take 15 mins, therefore the user will be able to see the updated capacity due to a live event that has not affected the simulation yet, thus being led to wrong conclusions. 

Having said that, I agree that capacity values could potentially be populated with data from the DB, and it is not necessary that we send the capacity values from d3a. This part of the code is mainly called by the simulation_results service, which has access to the DB, therefore it is feasible to read the capacity values directly from the DB. Also, since this function is triggered after the market slot has finished, we will not fall into the pitfall that I explained in the first paragraph. 

Of course, the interface of the `def update(self, area_result_dict, core_stats, current_market_slot):` method will need to be adapted to also incorporate one more argument for the latest representation from the ConfigurationSettings, however it is not that hard to adapt. An alternative approach would be to create a new method that will calculate only values based on the ConfigurationSettings (we could also include it in the abstractmethods in case it is useful for other simulation results in the future, but I would avoid it for now). 

All of the above apply to the rest of the comments below. I also hope that I do not start an endless discussion with this comment :(

Reply:
OK, understood and convinced with `However, for CNs, this process might take 15 mins, therefore the user will be able to see the updated capacity due to a live event that has not affected the simulation yet, thus being led to wrong conclusions.`
I also would vote for `An alternative approach would be to create a new method that will calculate only values based on the ConfigurationSettings (we could also include it in the abstractmethods in case it is useful for other simulation results in the future, but I would avoid it for now).`
Yeah, lets not drag this endlessly here. :-)

Reply:
@spyrostz how do you suggest adding the extra config_obj parameter to the update interface of my new handler class without adding it to the all_results's ResultHandler update method ? AFAIS this function invokes other update methods